airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/cdc/MongoDbDebeziumStateUtil.java

```diff
@@ -129,6 +121,46 @@ public boolean isValidResumeToken(final BsonDocument savedOffset,
     }
   }
 
+  public boolean isValidResumeToken(final BsonDocument savedOffset,
+                                    final MongoClient mongoClient,
+                                    final List<String> databaseNames,
+                                    final List<List<ConfiguredAirbyteStream>> streamsByDatabase) {
+    if (Objects.isNull(savedOffset) || savedOffset.isEmpty()) {
+      return true;
+    }
+
+    // databaseNames and streamsByDatabase must be the same length
```

Comment:
Can there be an edge case for database added to config but no stream selected in catalog?